extension/test/functional/tests/Infobar.spec.ts

```diff
@@ -0,0 +1,172 @@
+      // ... this test continues the session from the previous test
+      await navigateToURL(driver, fixtureUrl);
+      await assertInfobarIsShown(tabsCurrentlyOpened);
+      const translateButtonElement = await lookForInfobarTranslateButton(
+        driver,
+        tabsCurrentlyOpened,
+      );
+      assertElementExists(translateButtonElement, "translateButtonElement");
+      await takeScreenshot(driver, this.ctx.test.fullTitle());
+    });
+
+    const translateViaInfobar = async nthTab => {
+      const originalPageElement = await lookForFixturePageOriginalContent(
+        driver,
+      );
+      assertElementExists(originalPageElement, "originalPageElement");
+      await assertInfobarIsShown(tabsCurrentlyOpened);
+      const translateButtonElement = await lookForInfobarTranslateButton(
+        driver,
+        nthTab,
+      );
+      assertElementExists(translateButtonElement, "translateButtonElement");
+      await translateButtonElement.click();
+    };
+
+    const assertTranslationSucceeded = async () => {
+      const translatedPageElement = await lookForFixturePageTranslatedContent(
+        driver,
+        (maxToleratedModelLoadingDurationInSeconds +
+          maxToleratedTranslationDurationInSeconds) *
+          1000,
+      );
+      assertElementExists(translatedPageElement, "translatedPageElement");
+    };
+
+    it("Translation via the infobar works", async () => {
+      // ... this test continues the session from the previous test
+      await translateViaInfobar(tabsCurrentlyOpened);
+      await assertTranslationSucceeded();
+      await takeScreenshot(driver, this.ctx.test.fullTitle());
+    });
+
+    it("Translation via the infobar works after further navigations", async () => {
+      // ... this test continues the session from the previous test
+      await navigateToURL(driver, fixtureUrl);
+      await translateViaInfobar(tabsCurrentlyOpened);
+      await assertTranslationSucceeded();
+      await takeScreenshot(
+        driver,
+        `${this.ctx.test.fullTitle()} - After navigation 1`,
+      );
+      await navigateToURL(driver, fixtureUrl);
+      await translateViaInfobar(tabsCurrentlyOpened);
+      await assertTranslationSucceeded();
+      await takeScreenshot(
+        driver,
+        `${this.ctx.test.fullTitle()} - After navigation 2`,
+      );
+    });
+
+    it("Translation via the infobar works when translating in multiple tabs at the same time", async () => {
+      // ... this test continues the session from the previous test
+      await driver.switchTo().newWindow("tab");
+      tabsCurrentlyOpened++;
+      await navigateToURL(driver, fixtureUrl);
+      await translateViaInfobar(tabsCurrentlyOpened);
+      const originalWindow = await driver.getWindowHandle();
+      await driver.switchTo().newWindow("tab");
+      tabsCurrentlyOpened++;
+      await navigateToURL(driver, fixtureUrl);
```

Comment:
maybe we should add a case of translation of different language pairs on different tabs? we can add it later of course

Reply:
This is a production use case though, which might be quite tricky not only from models and memory perspective. There are some things that live in the chrome context at the moment but supposed to be in the content one, so it would be great to be able to catch possible problems in CI. Eventually.

Reply:
> maybe we should add a case of translation of different language pairs on different tabs? we can add it later of course
Yes, that is a great idea. I'll make sure to include it as part of the fix to https://github.com/mozilla-extensions/bergamot-browser-extension/issues/49 since it specifically treats such a case. 

> This is possible but for memory reasons we may not want to have two models loaded at once.
Exactly. Currently it will unload the current model if a request for another model is made, leading to a stable amount of memory used, but an extra delay in form of loading the other model.